Source/Iterators/Common/Identity.php

```diff
@@ -45,7 +45,9 @@ public static function hash($value)
 
             case 'o': //object
 
-                return 'o' . spl_object_hash($value);
+                return $value instanceof \DateTimeInterface || $value instanceof \DateTime
+                        ? 'dt:' . get_class($value) . ':' . $value->getTimestamp()
```

Comment:
What happens for other value types? For instance, we kept rejecting this particular sort of feature throughout doctrine because there is no way to build a "special cases" map.

While I agree that it makes developer experience better, it also causes a series of #WTF moments.

What if you could simply have the concept of "value objects" and a map of which ones are VOs? Acceptable? Better? Worse?